### DISCUSSION TITLE
scalability of frigate

### DISCUSSION THREAD

**Discussion by Zhanghyi** [blakeblackshear/frigate#6559] [General]
I haven't seen any discussions on scalability of frigate. If I want to apply it for commercial use with about 1000 cameras, what would the performance and hardware requirements be like? And how is the performance of frigate compared with other platforms such as zoneminder and shinobi(for fairness, we can discuss the performance of recording and live stream)?

**Comment by bazylhorsey** [comment]
Unlike Shinobi, there is no built in mesh. 
The most realistic way to do this would be multiple computers distributing camera ownership. Then using a singular Mqtt network and separating by client_id.
Each frigate container would have its own sql database and page. But you could make a HA automation OR a new consolidation app that watches MQTT and adds them to another database with all the combined events.
If these are among multiple facilities, I theorize something like AWS IoT core could be leveraged to bring everything together. But I have not seen anyone accomplish that.

If it's purely on prem you could keep a single network device running DeepStack on an expensive graphics card and then have a bunch of cheap edge devices close to the cameras.

**Comment by Sceptersax** [comment]
I also have a usecase for a large deployment camera system. What is currently the max number of cameras? If it's sub 100, is their anything on the road map for larger?

**Reply by NickM-27** [reply]
The only max is the theoretical max of your hardware, there is no hard limit

**Reply by Sceptersax** [reply]
That sounds great but if you don't mind may i ask what the largest recorded deployment has been to date?

**Reply by NickM-27** [reply]
We don't have analytics so there is no way to know for sure. There are multiple users here with 40+ camera installations and some users on discord with ~70 or so

**Reply by Sceptersax** [reply]
Perfect. Thank you
